--- elfutils-0.179/tests/run-lfs-symbols.sh.orig	2020-04-30 19:56:23.935187505 +0200
+++ elfutils-0.179/tests/run-lfs-symbols.sh	2020-04-30 19:57:13.321586623 +0200
@@ -48,7 +48,7 @@
 testrun_lfs() {
   echo "checking $1"
   if [ -e "$1" ]; then
-    bad=$(testrun ${abs_top_builddir}/src/nm -u "$1" | awk "$LFS")
+    bad=$(testrun ${abs_top_builddir}/src/nm -u "$1" | gawk "$LFS")
     if [ -n "$bad" ]; then
       echo "$1 contains non-lfs symbols:" $bad
       exit_status=1
