--- elfutils-0.137/src/addr2line.c.orig	2008-08-09 06:06:44.000000000 +0200
+++ elfutils-0.137/src/addr2line.c	2008-09-09 22:15:16.304992337 +0200
@@ -437,6 +437,12 @@
   return false;
 }
 
+#if (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 7) && (__GNUC__ >= 4) && (__GNUC_MINOR__ >= 3)
+# define SCANF_ADDR "%m"
+#else
+# define SCANF_ADDR "%a"
+#endif
+
 static int
 handle_address (const char *string, Dwfl *dwfl)
 {
@@ -447,10 +453,10 @@
       bool parsed = false;
       int n;
       char *name = NULL;
-      if (sscanf (string, "(%m[^)])%" PRIiMAX "%n", &name, &addr, &n) == 2
+      if (sscanf (string, "(" SCANF_ADDR "[^)])%" PRIiMAX "%n", &name, &addr, &n) == 2
 	  && string[n] == '\0')
 	parsed = adjust_to_section (name, &addr, dwfl);
-      else if (sscanf (string, "%m[^-+]%" PRIiMAX "%n", &name, &addr, &n) == 2
+      else if (sscanf (string, SCANF_ADDR "[^-+]%" PRIiMAX "%n", &name, &addr, &n) == 2
 	       && string[n] == '\0')
 	{
 	  /* It was symbol+offset.  */
