--- elfutils-0.164/tests/backtrace-data.c.orig	2015-10-24 16:54:40.223734605 +0200
+++ elfutils-0.164/tests/backtrace-data.c	2015-10-24 16:54:43.840589170 +0200
@@ -101,7 +101,7 @@
     {
       // 37e3c22000-37e3c23000 rw-p 00022000 00:11 49532 /lib64/ld-2.14.90.so */
       unsigned long start, end, offset;
-      i = fscanf (f, "%lx-%lx %*s %lx %*x:%*x %*x", &start, &end, &offset);
+      i = fscanf (f, "%lx-%lx %*s %lx %*x:%*x %*u", &start, &end, &offset);
       assert (errno == 0);
       assert (i == 3);
       char *filename = strdup ("");
