--- elfutils-0.178/libdw/libdwP.h.orig	2019-12-01 18:40:54.893054665 +0100
+++ elfutils-0.178/libdw/libdwP.h	2019-12-01 18:43:25.465572277 +0100
@@ -234,7 +234,11 @@
     size_t size;
     size_t remaining;
     struct libdw_memblock *prev;
-    char mem[0];
+    char mem[0]
+#ifdef __sparc__
+      __attribute__((aligned(8)))
+#endif
+      ;
   } **mem_tails;
 
   /* Default size of allocated memory blocks.  */
