--- elfutils-0.123/tests/run-native-test.sh.orig	2006-04-13 17:04:57.000000000 +0200
+++ elfutils-0.123/tests/run-native-test.sh	2006-08-25 22:55:40.053239000 +0200
@@ -58,7 +58,7 @@
 for cc in "$HOSTCC" "$HOST_CC" cc gcc "$CC"; do
   test "x$cc" != x || continue
   $cc -o native -g native.c > /dev/null 2>&1 &&
-  ./native > /dev/null 2>&1 & native=$! &&
+  { ./native > /dev/null 2>&1 & } && native=$! &&
   sleep 1 && kill -0 $native 2> /dev/null &&
   break ||
   native=0
