--- elfutils-0.178/configure.ac.orig	2019-11-26 23:48:42.000000000 +0100
+++ elfutils-0.178/configure.ac	2019-12-01 19:30:47.593508484 +0100
@@ -683,7 +683,7 @@
     AC_MSG_NOTICE([checking debuginfod dependencies, disable to skip])
     enable_debuginfod=yes # presume success
     PKG_PROG_PKG_CONFIG
-    if test "x$ac_cv_prog_ac_ct_CXX" = "x"; then enable_debuginfod=no; fi
+    if test "x$CXX" = "x"; then enable_debuginfod=no; fi
     PKG_CHECK_MODULES([libmicrohttpd],[libmicrohttpd >= 0.9.33],[],[enable_debuginfod=no])
     PKG_CHECK_MODULES([libcurl],[libcurl >= 7.29.0],[],[enable_debuginfod=no])
     PKG_CHECK_MODULES([sqlite3],[sqlite3 >= 3.7.17],[],[enable_debuginfod=no])
