--- elfutils-0.141/configure.ac#	2009-05-02 14:13:01.000000000 +0200
+++ elfutils-0.141/configure.ac	2009-05-02 14:12:53.000000000 +0200
@@ -290,7 +290,7 @@
 
 # 1.234<whatever> -> 1234<whatever>
 case "$PACKAGE_VERSION" in
-[[0-9]].*) eu_version="${PACKAGE_VERSION/./}" ;;
+[[0-9]].*) eu_version="${PACKAGE_VERSION%%.*}${PACKAGE_VERSION#*.}" ;;
 *)     	   AC_MSG_ERROR([confused by version number '$PACKAGE_VERSION']) ;;
 esac
 case "$eu_version" in
@@ -319,6 +319,6 @@
 esac
 
 # Round up to the next release API (x.y) version.
-[eu_version=$[($eu_version + 999) / 1000]]
+eu_version=$((($eu_version + 999) / 1000))
 
 AC_OUTPUT
