--- elfutils-0.155/tests/Makefile.am.orig	2012-08-27 20:27:31.000000000 +0200
+++ elfutils-0.155/tests/Makefile.am	2013-03-26 18:52:59.660956629 +0100
@@ -187,8 +187,7 @@
 if STANDALONE
 TESTS_ENVIRONMENT = $(installed_TESTS_ENVIRONMENT)
 else !STANDALONE
-TESTS_ENVIRONMENT = LC_ALL=C LANG=C $(srcdir)/test-wrapper.sh \
-		    ../libdw:../backends:../libelf:../libasm
+TESTS_ENVIRONMENT = LC_ALL=C LANG=C built_library_path=../libdw:../backends:../libelf:../libasm elfutils_testrun=built
 
 installcheck-local:
 	$(MAKE) $(AM_MAKEFLAGS) \
