diff -Nur elfutils-0.127.org/libdwfl/Makefile.am elfutils-0.127/libdwfl/Makefile.am
--- elfutils-0.127.org/libdwfl/Makefile.am	2007-06-07 15:44:39.000000000 +0000
+++ elfutils-0.127/libdwfl/Makefile.am	2007-06-07 15:44:58.474856486 +0000
@@ -33,7 +33,7 @@
 else
 AM_CFLAGS =
 endif
-AM_CFLAGS += -Wall -Werror -Wshadow -Wunused -Wformat=2 $(WEXTRA) -std=gnu99
+AM_CFLAGS += -Wall  -Wshadow -Wunused -Wformat=2 $(WEXTRA) -std=gnu99
 INCLUDES = -I. -I$(srcdir) -I$(srcdir)/../libelf -I$(srcdir)/../libebl \
 	   -I$(srcdir)/../libdw -I.. -I$(srcdir)/../lib
 VERSION = 1
diff -Nur elfutils-0.127.org/tests/Makefile.am elfutils-0.127/tests/Makefile.am
--- elfutils-0.127.org/tests/Makefile.am	2007-06-07 16:00:49.000000000 +0000
+++ elfutils-0.127/tests/Makefile.am	2007-06-07 16:01:03.268761805 +0000
@@ -27,11 +27,11 @@
 DEFS = -DHAVE_CONFIG_H -D_GNU_SOURCE
 WEXTRA = @WEXTRA@
 if MUDFLAP
-AM_CFLAGS = -Wall -Werror $(WEXTRA) -std=gnu99 -fmudflap\
+AM_CFLAGS = -Wall  $(WEXTRA) -std=gnu99 -fmudflap\
 	    $(if $($(*F)_no_Wformat),-Wno-format,-Wformat=2)
 BUILD_RPATH = \$$ORIGIN/../backends
 else
-AM_CFLAGS = -Wall -Werror $(WEXTRA) -std=gnu99 \
+AM_CFLAGS = -Wall  $(WEXTRA) -std=gnu99 \
 	    $(if $($(*F)_no_Wformat),-Wno-format,-Wformat=2)
 BUILT_RPATH = \$$ORIGIN/../libasm:\$$ORIGIN/../libdw:\$$ORIGIN/../backends:\$$ORIGIN/../libelf
 endif
--- elfutils-0.137/src/Makefile.am~orig	2008-11-23 22:27:16.000000000 +0100
+++ elfutils-0.137/src/Makefile.am	2008-11-23 22:35:06.000000000 +0100
@@ -33,7 +33,6 @@
 AM_CFLAGS =
 endif
 AM_CFLAGS += -Wall -Wshadow -std=gnu99 $(native_ld_cflags) \
-	     $(if $($(*F)_no_Werror),,-Werror) \
 	     $(if $($(*F)_no_Wunused),,-Wunused $(WEXTRA)) \
 	     $(if $($(*F)_no_Wformat),-Wno-format,-Wformat=2) $(CFLAGS_$(*F))
 
--- elfutils-0.141/libcpu/Makefile.am~	2009-05-02 14:18:41.000000000 +0200
+++ elfutils-0.141/libcpu/Makefile.am	2009-05-02 15:01:39.000000000 +0200
@@ -33,8 +33,7 @@
 WEXTRA = @WEXTRA@
 AM_CFLAGS += -Wall -Wshadow -Wunused $(WEXTRA) -std=gnu99 -fpic \
 	     -fdollars-in-identifiers \
-	     $($(*F)_CFLAGS) \
-	     $(if $($(*F)_no_Werror),,-Werror)
+	     $($(*F)_CFLAGS)
 INCLUDES = -I$(srcdir) -I$(srcdir)/../lib -I$(srcdir)/../libelf \
 	   -I$(srcdir)/../libebl -I$(srcdir)/../libdw -I$(srcdir)/../libasm
 LEXCOMPILE = $(LEX) $(LFLAGS) $(AM_LFLAGS) -P$(<F:lex.l=)
