--- elfutils-0.185/debuginfod/debuginfod-client.c.orig	2021-05-22 20:25:24.000000000 +0200
+++ elfutils-0.185/debuginfod/debuginfod-client.c	2021-06-11 19:22:31.611417051 +0200
@@ -267,7 +267,11 @@ debuginfod_init_cache (char *cache_path,
   if (fd < 0)
     return -errno;
 
+#if defined(__x86_64__) && defined(__ILP32__)
+  if (dprintf(fd, "%lld", cache_clean_default_interval_s) < 0)
+#else
   if (dprintf(fd, "%ld", cache_clean_default_interval_s) < 0)
+#endif
     return -errno;
 
   /* init max age config file.  */
@@ -275,7 +279,11 @@ debuginfod_init_cache (char *cache_path,
       && (fd = open(maxage_path, O_CREAT | O_RDWR, DEFFILEMODE)) < 0)
     return -errno;
 
+#if defined(__x86_64__) && defined(__ILP32__)
+  if (dprintf(fd, "%lld", cache_default_max_unused_age_s) < 0)
+#else
   if (dprintf(fd, "%ld", cache_default_max_unused_age_s) < 0)
+#endif
     return -errno;
 
   return 0;
