--- elfutils-0.159/tests/Makefile.am~	2015-01-04 14:41:29.000000000 +0000
+++ elfutils-0.159/tests/Makefile.am	2015-01-04 16:08:31.699271017 +0000
@@ -102,8 +102,8 @@
 	run-readelf-mixed-corenote.sh run-dwfllines.sh \
 	run-dwfl-report-elf-align.sh run-addr2line-test.sh \
 	run-addr2line-i-test.sh run-varlocs.sh run-funcretval.sh \
-	run-backtrace-native.sh run-backtrace-data.sh run-backtrace-dwarf.sh \
-	run-backtrace-native-biarch.sh run-backtrace-native-core.sh \
+	run-backtrace-dwarf.sh \
+	run-backtrace-native-core.sh \
 	run-backtrace-native-core-biarch.sh run-backtrace-core-x86_64.sh \
 	run-backtrace-core-i386.sh run-backtrace-core-ppc.sh \
 	run-backtrace-core-s390x.sh run-backtrace-core-s390.sh \
